requests/models.py

```diff
@@ -331,6 +330,22 @@ def prepare_method(self, method):
         if self.method is not None:
             self.method = to_native_string(self.method.upper())
 
+    @staticmethod
+    def _get_idna_encoded_host(host):
+        try:
+            from .packages import idna
+        except ImportError:
+            # tolerate the possibility of downstream repackagers unvendoring `requests`
+            # For more information, read: packages/__init__.py
+            import idna
+            sys.modules['%s.idna' % __name__] = idna
```

Comment:
This does not work. `__name__` here is `requests.models`, not `requests.packages` which is what we *want*.

The function to import `idna` belongs in `requests/packages/__init__.py`, not here.